components/style/values/computed/length.rs

```diff
@@ -78,7 +78,8 @@ impl ComputeSquaredDistance for CalcLengthOrPercentage {
         // FIXME(nox): This looks incorrect to me, to add a distance between lengths
         // with a distance between percentages.
         Ok(
-            self.unclamped_length().compute_squared_distance(&other.unclamped_length())? +
+            self.unclamped_length().to_f64_px().compute_squared_distance(
+                &other.unclamped_length().to_f64_px())? +
```

Comment:
Why is it correct at all to mix lengths and percentages? Cf. the FIXME I left just above that change.